### PR TITLE
No need to use response methods returning a Future, if not used (part II)

### PR DESCRIPTION
**Pull request by franz1981**
Similarly to https://github.com/quarkusio/quarkus/pull/25577 this is saving creating unnecessary garbage (that will live long enough to bother GC, affect performance and force Netty write completions advertising events we are not interested to deal with).

### CODE REVIEW

**Comment by franz1981** [comment]
@geoand I would like to get the opinion of Vert-x team first, as per the mentioned PR.

**Comment by franz1981** [comment]
@tsegismont Hi Thomas! Do you think this can cause problems similar to 
https://github.com/quarkusio/quarkus/pull/25577?

File: independent-projects/resteasy-reactive/server/vertx/src/main/java/org/jboss/resteasy/reactive/server/vertx/ResteasyReactiveOutputStream.java [hunk 1]
```diff
@@ -80,7 +82,7 @@ Buffer createBuffer(ByteBuf data) {
 
     public void write(ByteBuf data, boolean last) throws IOException {
         if (last && data == null) {
-            request.response().end();
+            request.response().end(EMPTY_BUFFER, null);
```
Comment (tsegismont) [inline independent-projects/resteasy-reactive/server/vertx/src/main/java/org/jboss/resteasy/reactive/server/vertx/ResteasyReactiveOutputStream.java:85]:
```suggestion
            request.response().end((Handler<AsyncResult<Void>>) null);
```

You could save the empty buffer creation like this. Vert.x does it internally.

**Review by tsegismont — APPROVED** [review]
Fine with me but, as discussed elsewhere, this kind of optimization will no longer be effective when we upgrade to Netty 5 (void channel promise removal)

**Comment by franz1981** [comment]
Thanks @tsegismont ! Yep we're aware, but we don't know when will happen and the existing users can have this small benefit in the meantime (tiny, but still something :) )

**Review by geoand — APPROVED** [review]

**Pull request merged**